pkg/statedb/deletetracker.go

```diff
@@ -106,15 +110,28 @@ func (dt *DeleteTracker[Obj]) Process(txn ReadTxn, minRevision Revision, process
 			dt.Mark(rev - 1)
 
 			// Processing failed, stop here and try again from this same revision.
-			closedWatch := make(chan struct{})
-			close(closedWatch)
-			return rev, closedWatch, err
+			return closedWatchChannel, err
 		}
 
 	}
 
 	// Fully processed up to latest table revision. GC deleted objects
 	// and return the next revision.
 	dt.Mark(upTo)
-	return upTo + 1, watch, nil
+	return watch, nil
+}
+
+// Iterate over updated and deleted objects in revision order.
+func (dt *DeleteTracker[Obj]) Iterate(txn ReadTxn, iterateFn func(obj Obj, deleted bool, rev Revision)) <-chan struct{} {
```

Comment:
So `Iterate` is basically the non-failable version of `Process`. Perhaps we can come up with two names so that is clear? Now it seems like they might do completely different things.

Reply:
Ah didn't think of that. I dislike `Process` in general a bit as it's vague. WDYT about renaming it to `IterateWithError`?